### ISSUE TITLE
Laravel Executable Requires sudo

### ISSUE THREAD

**Issue by nomanurrehman** [laravel/homestead#1028]
Inside the homestead box, I have to "sudo laravel new ..." as "laravel new ..." halts with permission errors. This is a bit confusing since immediately after executing the command, composer says:

```bash
Do not run Composer as root/super user! See https://getcomposer.org/root for details
```

What is even more confusing is the file permissions are all correct.

## Versions
  - Vagrant: 2.2.3
  - Provider: Virtualbox
  - Homestead: 7.20.0

## Host operating system
Red Hat Enterprise Linux 7

## Homestead.yaml
```yaml
---
ip: "192.168.10.10"
memory: 3072
cpus: 2
provider: virtualbox
mariadb: true

authorize: ~/.ssh/id_rsa.pub

keys:
    - ~/.ssh/id_rsa

folders:
    - map: /home/noman/Development/PHP/voyager
      to: /home/vagrant/projects/voyager

sites:
    - map: voyager.dev
      to: /home/vagrant/projects/voyager/public
      php: "7.3"

databases:
    - voyager
```

## Expected behavior
Is this how it is supposed to work and there is nothing to be alarmed about?

## Actual behavior
I think only doing "laravel new ..." should work.

## Steps to reproduce
Simply provision a homestead box and try creating a new project inside "home/vagrant/code" using "laravel new".

**Comment by nomanurrehman** [comment]
Okay, I might have misunderstood how to use it and got a little impatient with the issue creation. Sorry, everyone!

**Comment by nomanurrehman** [comment]
Okay, this is still a problem for me. I just setup a new machine and had to do `sudo` to create a new laravel project. Details are in the first post of this thread.

**Comment by roberto-aguilar** [comment]
Can you please share the output of your `vagrant destroy && vagrant up`?

I have tried this in a fresh installation with the latest box and i can run `laravel new ...` without it requiring sudo.

**Comment by svpernova09** [comment]
You really *shouldn't* be creating new Laravel sites from *inside* Homestead. One of the principles of Vagrant, one of the tools behind Homestead is that you shouldn't put *anything in the vm* via. manual process. Meaning you never create things *in* the VM that you care about. The reason for this principle is so that you never lose work by destroying your virtual machine.

If you *must* create your projects *inside* Homestead here is the best way to ensure you don't lose any of your work:

* Install Homestead and run `vagrant up` (Skip adding any sites for now)
* `vagrant ssh`
* `laravel new <PROJECT-NAME>`
* `mv <PROJECT-NAME> /vagrant/.`

On your *HOST* machine: Open the folder you ran `vagrant up` in and _move_ the <PROJECT-NAME> to wherever you keep your projects (Mine is `/~Code`).

* Update your `Homestead.yaml` and add a folder path to where you moved <PROJECT-NAME> and add a site for the folder to the configuration as well.
* `vagrant destroy && vagrant up` 

Your <PROJECT-NAME> will be mapped from it's location on your host OS to the Homestead VM.

**Comment by nomanurrehman** [comment]
@svpernova09 

Okay, understood! I have thoroughly read the Laravel documentation but it does not mention anywhere how to create a Laravel project inside the VM without touching it. Not that I am criticising the documentation but can you point me in the right direction?

Thanks.

**Comment by stefkes** [comment]
@nomanurrehman the issue you're facing is probably because you're working directly in the VM (it was the case for me).

To create a project outside the VM you just do this from your local file system. cd to the dir where you keep your projects (ex `/Users/nomanurrehman/Sites/`) and there you do `laravel new project-name
`
Then edit your `Homestead.yaml` file to add an entry among the "folders" to sync this project-name directory to your VM.

After editing Homestead.yaml you do a `vagrant reload --provision`

**Comment by nomanurrehman** [comment]
@stefkes Couple of years ago when I started using Vagrant, I think I get the idea of what was the issue back then.

Nevertheless, thank you for taking the time to post such a detailed reply.

I hope this thread helps someone who is just starting out like I was.

**Comment by itsjustanthony** [comment]
> @nomanurrehman the issue you're facing is probably because you're working directly in the VM (it was the case for me).
> 
> To create a project outside the VM you just do this from your local file system. cd to the dir where you keep your projects (ex `/Users/nomanurrehman/Sites/`) and there you do `laravel new project-name `
> 
> Then edit your `Homestead.yaml` file to add an entry among the "folders" to sync this project-name directory to your VM.
> 
> After editing Homestead.yaml you do a `vagrant reload --provision`

This workflow requires me to install all of the requirements for laravel on every new host machine I want to develop from. 

I thought a goal of using homestead was to prevent that exact kind of setup from being necessary?

I shouldn't have to install the fully supported stack on my HOST in order to develop with a virtual machine, that literally ruins a major point of using a virtual machine for development.

**Comment by nomanurrehman** [comment]
@itsjustanthony Yes, these were my exact thoughts when I started using it.

What really bugged me was some parts need to be on my machine in order to use the others parts that I can setup inside the virtual machine.

I would highly recommend you look into [Docker](https://www.docker.com). That thing has completely changed how I used to do development. Also, you have the added benefit of learning something that a lot of companies use in their development stack, unlike Vagrant. By that, I mean the packaging and deployment part.
